docs/presentations/linkedin-meetup-2024-04-24.qmd

```diff
@@ -0,0 +1,549 @@
+
+::: {.panel-tabset}
+## DuckDB
+
+```python
+import ibis
+
+con = ibis.duckdb.connect()
+penguins = con.table("penguins")
+penguins.group_by("species", "island").agg(count=ibis._.count())
+```
+
+An embeddable, zero-dependency, C++ SQL database engine.
+
+## DataFusion {auto-animate="true"}
+
+```python
+import ibis
+
+con = ibis.datafusion.connect()
+penguins = con.table("penguins")
+penguins.group_by("species", "island").agg(count=ibis._.count())
+```
+
+A Rust SQL query engine.
+
+## ClickHouse {auto-animate="true"}
+
+```python
+import ibis
+
+con = ibis.clickhouse.connect()
+penguins = con.table("penguins")
+penguins.group_by("species", "island").agg(count=ibis._.count())
+```
+
+A C++ column-oriented database management system.
+
+## Polars {auto-animate="true"}
+
+```python
+import ibis
+
+con = ibis.polars.connect()
+penguins = con.table("penguins")
+penguins.group_by("species", "island").agg(count=ibis._.count())
+```
+
+A Rust DataFrame library.
+
+## BigQuery {auto-animate="true"}
+
+```python
+import ibis
+
+con = ibis.bigquery.connect()
+penguins = con.table("penguins")
+penguins.group_by("species", "island").agg(count=ibis._.count())
+```
+
+A serverless, highly scalable, and cost-effective cloud data warehouse.
+
+## Snowflake {auto-animate="true"}
+
+```python
+import ibis
+
+con = ibis.snowflake.connect()
+penguins = con.table("penguins")
+penguins.group_by("species", "island").agg(count=ibis._.count())
+```
+
+A cloud data platform.
+
+## Oracle {auto-animate="true"}
+
+```python
+import ibis
+
+con = ibis.oracle.connect()
+penguins = con.table("penguins")
+penguins.group_by("species", "island").agg(count=ibis._.count())
+```
+
+A relational database management system.
+
+## Spark {auto-animate="true"}
+
+```python
+import ibis
+
+con = ibis.pyspark.connect()
+penguins = con.table("penguins")
+penguins.group_by("species", "island").agg(count=ibis._.count())
+```
+
+A unified analytics engine for large-scale data processing.
+
+## Trino {auto-animate="true"}
+
+```python
+import ibis
+
+con = ibis.trino.connect()
+penguins = con.table("penguins")
+penguins.group_by("species", "island").agg(count=ibis._.count())
+```
+
+A distributed SQL query engine.
+
+## Flink {auto-animate="true"}
+
+```python
+import ibis
+
+con = ibis.flink.connect()
+penguins = con.table("penguins")
+penguins.group_by("species", "island").agg(count=ibis._.count())
+```
+
+A distributed streaming and batch SQL analytics engine.
+:::
+
+# How does Ibis work?
+
+This is going to be very fast 🏃💨. I will happily answer questions about it 😂.
+
+## Components
+
+- Expression system
+- Compiler
+- Driver
+
+## Expressions
+
+- Expr vs op
+- Type system
+- Pattern matching
+- Traversal
+
+## Expressions cont.
+
+```{python}
+#| echo: true
+expr = (
+  pengs.group_by("species")
+  .mutate(s.across(s.numeric() & ~s.c("year"), (_ - _.mean()) / _.std()))
+  .head(5)
+)
+```
+
+Graph representation
+
+```{python}
+from ibis.expr.visualize import to_graph
```

Comment:
this code is not shown and I think it should be personally

Reply:
Can do.

Reply:
Done.